src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -145,11 +145,22 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             $this->loaders = new \ArrayIterator();
 
             try {
-                while ((false === $env || null === $env) && $loaders->valid()) {
-                    $loader = $loaders->current();
-                    $loaders->next();
+                $index = 0;
+                $ended = true;
+                foreach ($loaders as $loader) {
+                    if ($index++ < $this->loadedLoaderIndex) {
+                        continue;
+                    }
                     $this->loadedVars[] = $vars = $loader->loadEnvVars();
+                    $this->loadedLoaderIndex = $index;
```

Comment:
shouldn't this be moved one line up, so that we don't call `loadEnvVars()` twice when a loader fails?

Reply:
I was thinking about a ParameterCircularReferenceException triggered when calling `loadEnvVars`. But it shouldn't be the case.
Fixed